src/modules/cubedos-file_server-api.ads

```diff
@@ -54,91 +55,240 @@ package CubedOS.File_Server.API is
-
-   function Read_Request_Encode
-     (Sender_Address : in Message_Address;
-      Request_ID     : in Request_ID_Type;
-      Handle         : in Valid_File_Handle_Type;
-      Amount         : in Read_Size_Type;
-      Priority       : in System.Priority := System.Default_Priority) return Message_Record
-     with Global => null;
-
-   function Read_Reply_Encode
-     (Receiver_Address : in Message_Address;
-      Request_ID       : in Request_ID_Type;
-      Handle           : in Valid_File_Handle_Type;
-      Amount           : in Read_Result_Size_Type;
-      Data             : in CubedOS.Lib.Octet_Array;
-      Priority         : in System.Priority := System.Default_Priority) return Message_Record
-     with
-       Global => null,
-       Pre => Amount <= Data'Length;
-
-   function Write_Request_Encode
-     (Sender_Address : in Message_Address;
-      Request_ID     : in Request_ID_Type;
-      Handle         : in Valid_File_Handle_Type;
-      Amount         : in Write_Size_Type;
-      Data           : in CubedOS.Lib.Octet_Array;
-      Priority       : in System.Priority := System.Default_Priority) return Message_Record
-     with
-       Global => null,
-       Pre => Amount <= Data'Length;
-
-   function Write_Reply_Encode
-     (Receiver_Address : in Message_Address;
-      Request_ID       : in Request_ID_Type;
-      Handle           : in Valid_File_Handle_Type;
-      Amount           : in Write_Result_Size_Type;
-      Priority         : in System.Priority := System.Default_Priority) return Message_Record
-     with Global => null;
-
-   function Close_Request_Encode
-     (Sender_Address : in Message_Address;
-      Request_ID     : in Request_ID_Type;
-      Handle         : in Valid_File_Handle_Type;
-      Priority       : in System.Priority := System.Default_Priority) return Message_Record
-     with Global => null;
-
-
-
-   function Is_Open_Request(Message : in Message_Record) return Boolean is
-     (Message_Manager.Message_Type(Message) = (This_Module, Message_Type'Pos(Open_Request)));
-
-   function Is_Open_Reply(Message : in Message_Record) return Boolean is
-     (Message_Manager.Message_Type(Message) = (This_Module, Message_Type'Pos(Open_Reply)));
-
-   function Is_Read_Request(Message : in Message_Record) return Boolean is
-     (Message_Manager.Message_Type(Message) = (This_Module, Message_Type'Pos(Read_Request)));
-
-   function Is_Read_Reply(Message : in Message_Record) return Boolean is
-     (Message_Manager.Message_Type(Message) = (This_Module, Message_Type'Pos(Read_Reply)));
-
-   function Is_Write_Request(Message : in Message_Record) return Boolean is
-     (Message_Manager.Message_Type(Message) = (This_Module, Message_Type'Pos(Write_Request)));
-
-   function Is_Write_Reply(Message : in Message_Record) return Boolean is
-     (Message_Manager.Message_Type(Message) = (This_Module, Message_Type'Pos(Write_Reply)));
-
-   function Is_Close_Request(Message : in Message_Record) return Boolean is
```

Comment:
Speak about the switch to universal message type

Talk about the new module ID semantics